torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -212,8 +213,8 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
             # for PP, DP, TP case, dp mesh dim would be 1, tp mesh dim would be 2
             # DP/TP would only live in the inner most 2-3 dims (HSDP + TP would be 3)
             dp_tp_mesh_ndim = dp_mesh.ndim + tp_mesh.ndim
-            outer_mesh_ndim = self._global_mesh.ndim - dp_tp_mesh_ndim
-            if self._global_mesh.ndim > dp_tp_mesh_ndim:
+            outer_mesh_ndim = dp_global_mesh.ndim - dp_tp_mesh_ndim
```

Comment:
i'm kinda confused now.  This PR actually does fix my torchtitan/3D issue.  But i wonder why we need to maintain the 'global_*' dims, couldn't we just use the spmd submesh for everything?

Reply:
Do you mean `_global_mesh`? FSDP need this to construct the DTensor for state_dict and parameter, iirc.

Reply:
no, I refer to self._global_mesh in the comment above. I propose to keep this variable but rename it to self._spmd_submesh since it is not actually global. 

then, I notice that while we are using _spmd_submesh for tensor creation, we seem to still use 'global' indices for sharing plan.  It looks like a mismatch to me.

But I think it's actually a bad naming?  `global_dp_mesh_dim` actually started out as global but then we subtract `outer_mesh_ndim` which I think makes it into a `local_dp_mesh_dim` instead, but we don't change its name.  and then we store placements in `global_placements` but again, its not really global, its 'relative to the SPMD submesh'.  

Does that make sense?

Reply:
or put another way, I think these variables have semantically confusing names and I wanted to get some suggestions how to improve them, or maybe add a clarifying comment if folks think they do make sense and I'm just missing the point.

`global_*_mesh_dim`
`global_placements`
`global_*`